jobs/e2e-test.groovy

```diff
@@ -290,6 +293,28 @@ def runTestServer() {
          runSmokeTestsArgs += ["--skip-tests=${params.SKIP_TESTS}"];
       }
 
+      // Determine if we actually need to run any tests at all
+      // tests = sh(script: "testing/runtests_server.py -n ${exec.shellEscapeList(runSmokeTestsArgs)} . ", returnStdout: true);
+      tests = exec.outputOf(["testing/runtests_server.py", "-n", runSmokeTestsArgs, "."].flatten());
+
+      // The runtests_server.py script with -n outputs all tests to run of 
+      // various types. We only need to worry about smoke tests, which are 
+      // also the last section, so grab everything after the header. If it's 
+      // empty, there are no tests to run.
+      e2eTests = tests.substring(tests.lastIndexOf("SMOKE TESTS:") + 12)
```

Comment:
It may be worth noting this offset changed to 12 from 13. The 12 is the correct length of `SMOKE TESTS:`, 13 was just stripping a newline for us. That behavior prevented the `\n\n` from matching (since the first `\n` was removed) if there were no e2e tests and only a single line before a new section.

Reply:
I think `"SMOKE_TESTS:".length()` would be even clearer!